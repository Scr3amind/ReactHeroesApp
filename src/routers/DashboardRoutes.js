import React from 'react'
import { Redirect, Route, Switch } from 'react-router'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { Navbar } from '../components/ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <div >
            <div className="w-screen h-screen bg-comic-pattern bg-repeat bg-contain opacity-30 absolute z-0"></div>
            <Navbar />
            <div className="relative z-10">
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen} />
                    <Route exact path="/hero/:heroeId" component={HeroScreen} />
                    <Route exact path="/dc" component={DcScreen} />
                    <Route exact path="/search" component={SearchScreen} />

                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </div>
    )
}
