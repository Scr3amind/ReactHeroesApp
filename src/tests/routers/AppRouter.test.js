import { mount, shallow } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";
import "../../setupTest";

describe('Pruebas en <AppRouter />', () => {

    const contextValue = {
        dispatch : jest.fn(),
        user: {
            logged: false
        }
    }

    test('debe de mostrar login si no se está autenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el componente de marvel si está autenticado', () => {
        const contextValue = {
            dispatch : jest.fn(),
            user: {
                name: 'Isaac',
                logged: true
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect(wrapper.find('nav').exists()).toBe(true);
    })
    
    
    
})
