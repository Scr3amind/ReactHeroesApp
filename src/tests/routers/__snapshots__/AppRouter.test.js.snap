// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar login si no se est√° autenticado 1`] = `
<div>
  <div
    className="h-screen flex flex-col items-center justify-center bg-comic-pattern bg-repeat bg-contain "
  >
    <div
      className=" flex flex-col h-1/3 w-2/3 justify-evenly rounded-3xl shadow-2xl bg-gray-600 items-center animate__animated animate__bounceIn "
    >
      <h1
        className="text-center text-6xl text-white"
      >
        Login Screen
      </h1>
      <button
        className="text-white bg-gray-900 rounded-lg transition duration-500 ease-in-out transform hover:-translate-y-1 hover:scale-110 py-2 my-4 px-4"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
